R/load_pop_data.R

```diff
@@ -107,54 +104,51 @@ set_age <- function(object, ...){
 }
 
 #' @export
-set_age.pop_data <- function(object, age = 'Age', standardize, ...){
+set_age.pop_data <- function(object, age = 'Age', standardize = TRUE, ...){
 
-  if(standardize)
-  {
-    # set age attribute
-    attr(object, "age_var") <- 'age'
-
-    # rename provided column
-    if(age %in% colnames(object))
-    {
-      object <- object %>%
-        rename('age' = age)
-    } else
-    {
-      cli::cli_abort(
-        '`age = "{age}"` is not a valid input;
-      the age column in not available in the data set')
-    }
-
-
-  } else if (!standardize)
-  {
     # check if age column exists
     if(age %in% colnames(object))
     {
       attr(object, "age_var") <- age
     } else
     {
-      # search age variable from object
-      age_var <- object %>%
-        select(tidyselect::matches("age",colnames(object),ignore.case = TRUE) & ends_with("e", ignore.case = TRUE)) %>%
-        names()
 
-      if(length(age_var) > 0)
+      cli::cli_alert_warning('The specified `age` column "{age}" does not exist.')
+
+      # search age variable from object
+      age_var <-
+        grep(
+          x = colnames(object),
+          value = TRUE,
+          pattern = age,
+          ignore.case = TRUE
+          )
+
+      if(length(age_var) == 1)
       {
         attr(object, "age_var") <- age_var
 
         # create warning when using searched age instead of provided age
-        cli::cli_alert_warning('The provided age attribute "{age}" does not exist.
-                        Proceeding to use "{age_var}"')
+        cli::cli_alert_info('Proceeding to use "{age_var}"')
 
-      } else
+      } else if (length(age_var) == 0)
       {
-        cli::cli_abort(
-          '`age = "{age}"` is not a valid input;
-      the age column in not available in the data set')
+        cli::cli_abort('No similar column name was detected.')
+      } else # if (length(age_var) > 1)
```

Comment:
The `if` statement here is commented out because it's already implied by the previous two conditions, so it would waste a few milliseconds to keep it active; let's leave it as a comment.